### PR TITLE
RME Result Interface

### PR DESCRIPTION
**Pull request by DanTanAtAims**
Results from ReefModEngine.jl can be loaded with the `load_results` function. There are two options depending on the location of the results and accompanying data files. The datafiles in the RME directory can be used as the data directory.

```julia
rs = ADRIA.load_results(RMEResultSet, "<path to data dir>", "<path to results dir>")
```

Expected data directory structure.

```bash
data_dir
│
├───con_bin
│       CONNECT_ACRO_2010_11.bin
│       CONNECT_ACRO_2011_12.bin
│       CONNECT_ACRO_2012_13.bin
│       CONNECT_ACRO_2014_15.bin
│       CONNECT_ACRO_2015_16.bin
│       CONNECT_ACRO_2016_17.bin
│
├───id
│       id_list_2023_03_30.csv
│
└───region
       reefmod_gbr.gpkg
```

The results data directory should contain a `results.nc` NetCDF file and `scenarios.csv`. If
the path to the results directory is not supplied, the results are expected to be
contained in a results subdirectory.

```julia
rs = ADRIA.load_domain(RMEResultSet, "<path to data dir>")
```

Expected data directory structure for defaulted result directory.

```bash
data_dir
├───con_bin
├───id
├───region
└───results
```
Allows access to sensitivity analysis, some of ADRIA metrics and map visualization.

image
image

### CODE REVIEW

**Comment by ConnectedSystems** [comment]
Hey @DanTanAtAims is this replacing #721 ?

**Comment by DanTanAtAims** [comment]
> Hey @DanTanAtAims is this replacing #721 ?

Yes, however the C-Scape-Result-Interface will be submitted separately. I'll delete the other pull request.

**Comment by DanTanAtAims** [comment]
@ConnectedSystems The results interface should now allow for nearly all analysis tools with toy data below. I should update ReefModEngine to export a more complete scenario specification beyond purely intervention factors.

## Time series clustering
RME_ts_clustering

## SIRUS
RME_SIRUS

## Temporal Sensitivity Analysis
RME_TSA

**Comment by ConnectedSystems** [comment]
Thanks Dan.

I suspect there's a simpler approach we could take compared to reimplementing methods for an RMEResultSet. I'll have to think about it.

File: docs/src/usage/results.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The results data directory should contain a `results.nc` NetCDF file and `scenarios.csv`. If
+the path to the results directory is not supplied, the results are expected to be
+contained in a results subdirectory.
+
+```julia
+rs = ADRIA.load_domain(RMEResultSet, "<path to data dir>")
+```
+
+Expected data directory structure for defaulted result directory.
```
Comment (ConnectedSystems) [inline docs/src/usage/results.md:40]:
```suggestion
Expected data directory structure for the default result directory is shown below.
```

File: docs/src/usage/results.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+## Loading ReefModEngine Results
+
+Results from ReefModEngine.jl can be loaded with the `load_results` function. There are two
+options depending on the location of the results and accompanying data files. 
+
+```julia
+rs = ADRIA.load_results(RMEResultSet, "<path to data dir>", "<path to results dir>")
+```
+
+Expected data directory structure.
```
Comment (ConnectedSystems) [inline docs/src/usage/results.md:12]:
```suggestion
Expected data directory structure:
```

File: src/ADRIA.jl [hunk 3]
```diff
@@ -66,6 +66,7 @@ include("interventions/fogging.jl")
 
 include("io/ResultSet.jl")
 include("io/result_io.jl")
+include("io/rme_result_io.jl")
```
Comment (ConnectedSystems) [inline src/ADRIA.jl:69]: Standardize file name.

Known / established acronyms should always be capitalized (e.g., rme -> RME)

File: docs/src/usage/results.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Loading Results
+
+## Loading ReefModEngine Results
+
+Results from ReefModEngine.jl can be loaded with the `load_results` function. There are two
+options depending on the location of the results and accompanying data files.
```
Comment (ConnectedSystems) [inline docs/src/usage/results.md:6]: You state "There are two options" but only mention show one? What is the other option?

File: src/io/rme_result_io.jl [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+    scenario_groups::Dict{Symbol, BitVector} = Dict()
+
+    # If the runs do not contain counterfactual or intervened runs exclude the key
+    if any(counterfactual_scens)
+        scenario_groups[:counterfactual] = counterfactual_scens
+    end
+    if any(intervened_scens)
+        scenario_groups[:intervened] = intervened_scens
+    end
+    return scenario_groups
```
Comment (ConnectedSystems) [inline src/io/rme_result_io.jl:194]: Same here: spacing for readability.

Note `return` statements, with very few exceptions, should always be separated for clarity.

```suggestion
    # If the runs do not contain counterfactual or intervened runs exclude the key
    if any(counterfactual_scens)
        scenario_groups[:counterfactual] = counterfactual_scens
    end

    if any(intervened_scens)
        scenario_groups[:intervened] = intervened_scens
    end

    return scenario_groups
```

**Review by ConnectedSystems — CHANGES_REQUESTED** [review]
Just a few more formatting related changes, but otherwise looks good to me!

**Comment by DanTanAtAims** [comment]
Rebased branch on to main before merge

**Review by ConnectedSystems — APPROVED** [review]
Thanks, looking good!

**Pull request merged**